src/components/docs/Header.astro

```diff
@@ -1,22 +1,19 @@
 ---
+import type { Props } from '@astrojs/starlight/props';
```

Comment:
This is the official method of doing Starlight component override, see https://starlight.astro.build/guides/overriding-components/ for documentation.